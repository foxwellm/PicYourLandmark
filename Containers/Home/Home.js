import React, { Component } from 'react';
import { StyleSheet, View, Text, ImageBackground, TouchableOpacity } from 'react-native';
import MapView, { Marker, Callout } from 'react-native-maps';
import { Icon, Image } from 'react-native-elements';

export class Home extends Component {
  constructor() {
    super()
    this.state = {
      clickedLocation: false,
      calculatedDistance: 0,
      selectedName: '',
      selectedVisited: false,
      selectedDescription: '',
      selectedPoints: 10,
      selectedID: -1,
      selectedLongitude: 0,
      selectedLatitude: 0,
      currentLocation: {
        currentLatDelta: 0.015,
        currentLonDelta: 0.0121
      },
      locations: [
        {
          id: 1,
          lat: 39.75302,
          lon: -104.9965,
          visited: false,
          name: 'Summit',
          description: 'example 1 description'
        },
        {
          id: 2,
          lat: 39.75023,
          lon: -104.9965,
          visited: true,
          name: 'The Delectable Egg',
          description: 'example 2 description'
        },
        {
          id: 3,
          lat: 39.77023,
          lon: -104.9965,
          visited: false,
          name: 'Far away example',
          description: 'over a mile away'
        }
      ]
    }
  }

  calculateDistance = ({ lat: landmarkLatitude, lon: landmarkLongitude, name, visited, description, id }) => {
    const earthRadiusMiles = 3958.8;
    const { currentLatitude, currentLongitude } = this.props;
    const deltaLatitude = (currentLatitude - landmarkLatitude) * Math.PI / 180;
    const deltaLongitude = (currentLongitude - landmarkLongitude) * Math.PI / 180;
    const landmarkLatitudeInRads = landmarkLatitude * Math.PI / 180;
    const currentLatitudeInRads = currentLatitude * Math.PI / 180;
    const a = Math.sin(deltaLatitude / 2) * Math.sin(deltaLatitude / 2) + Math.sin(deltaLongitude / 2) * Math.sin(deltaLongitude / 2) * Math.cos(landmarkLatitudeInRads) * Math.cos(currentLatitudeInRads);
    const c = 2 * Math.atan2(Math.sqrt(a), Math.sqrt(1 - a));
    const calculatedDistance = (earthRadiusMiles * c).toFixed(2);
    this.setState({
      calculatedDistance,
      selectedID: id,
      selectedName: name,
      selectedVisited: visited,
      selectedDescription: description,
      selectedLatitude: landmarkLatitude,
      selectedLongitude: landmarkLongitude,
      clickedLocation: true
    })
  }

  sendLocationPhoto = () => {
    const { selectedName, selectedDescription, selectedLatitude, selectedLongitude, selectedID } = this.state;
    this.props.takeLocationPhoto(selectedName, selectedDescription, selectedLatitude, selectedLongitude, selectedID)
  }

  render() {
    const { currentLatDelta, currentLonDelta } = this.state.currentLocation;
    const { currentLatitude, currentLongitude, allLocations, visitedLocationIds } = this.props;
    const { locations, clickedLocation, selectedName, selectedVisited, selectedPoints, calculatedDistance } = this.state;
    return (
      <View style={styles.container}>
        {
          clickedLocation && selectedVisited ? <ImageBackground source={require('../../assets/statueofliberty.jpg')} style={styles.imageBackground}>
            <View style={styles.overlay} />
            <View style={styles.flexRow}>
              <Text style={styles.headerText}>{selectedName}</Text>
            </View>
            <View style={styles.flexRow}>
              <Icon color="#f44336" name="diamond" type="font-awesome" size={15} />
              <Text style={[styles.pointsText, { color: 'white' }]}>{selectedPoints} gems</Text>
            </View>
            <View style={styles.flexRow}>
              <Icon color="#00bcd4" name="car" type="font-awesome" size={15} />
              <Text style={[styles.pointsText, { color: 'white' }]}>{calculatedDistance} mi. away </Text>
            </View>
            {
              calculatedDistance < 1 &&
              <View style={styles.flexRow}>
                <TouchableOpacity style={styles.locationCamera} onPress={() => this.sendLocationPhoto()}>
                  <Icon color="#4caf50" name="camera" type="font-awesome" size={15} marginTop={15} />
                  <Text style={[styles.pointsText, { color:'#c9c9c9', marginTop: 4}]}>Click to Retake Photo!</Text>
                </TouchableOpacity>
              </View>
            }
          </ImageBackground>
            : clickedLocation ?
              <ImageBackground source={require('../../assets/statueofliberty.jpg')} style={styles.imageBackground}>
                <View style={styles.overlay} />
                <View style={styles.flexRow}>
                  <Text style={styles.headerText}>{selectedName}</Text>
                </View>
                <View style={styles.flexRow}>
                  <Icon color="#f44336" name="diamond" type="font-awesome" size={15} />
                  <Text style={[styles.pointsText, { color: "white" }]}>{selectedPoints} gems</Text>
                </View>
                <View style={styles.flexRow}>
                  <Icon color="#00bcd4" name="car" type="font-awesome" size={15} />
                  <Text style={styles.pointsText}>{calculatedDistance} mi. away</Text>
                </View>
                {
                  calculatedDistance < 1 ?
                    <View style={styles.flexRow}>
                      <TouchableOpacity style={styles.locationCamera} onPress={() => this.sendLocationPhoto()}>
                        <Icon color="#4caf50" name="camera" type="font-awesome" size={15} marginTop={15}/>
                        <Text style={[styles.pointsText, { color: '#c9c9c9', marginTop: 4}]}>Click to Take Photo!</Text>
                      </TouchableOpacity>
                    </View>
                    :
                    <View>
                      <Text style={[styles.pointsText, { fontSize: 14, marginTop: 10, padding: 2, color: '#c9c9c9'}]}>Get within 1 mile of the landmark to take a photo and add it to your collection</Text>
                    </View>
                }
              </ImageBackground>
              :
              <ImageBackground source={require('../../assets/statueofliberty.jpg')} style={styles.imageBackground}>
                <View style={styles.overlay} />
                <Text style={styles.bannerText}>Earn gems by taking and uploading pics of you at various locations</Text>
                <Icon color="white" name="camera-retro" type="font-awesome" size={40} paddingTop={15}/>
              </ImageBackground>
        }
        {
          currentLatitude ?
            <MapView
              provider={"google"}
              style={styles.map}
              initialRegion={{
                latitude: currentLatitude,
                longitude: currentLongitude,
                latitudeDelta: currentLatDelta,
                longitudeDelta: currentLonDelta,
              }}
              showsUserLocation={true}
            >
              {
                allLocations.map(location => {
                  const pinColor = visitedLocationIds.includes(location.id) ? '#FFF000' : '#000FFF';
                  return (
                    <Marker
                      key={location.id}
                      coordinate={{
                        latitude: location.lat,
                        longitude: location.lon
                      }}
                      pinColor={pinColor}
                      onPress={() => {
                        this.calculateDistance(location);
                      }}
                    >
                      <Callout tooltip={true} >
                        <View style={styles.landmarkInfo}>
                          <Text style={{ fontWeight: 'bold' }}>{location.name}</Text>
                          <Text style={{ fontStyle: 'italic' }}>{location.description}</Text>
                          <Text>Distance away: mi</Text>
                          <Text>Visited: no</Text>
                          <Text>Link - takes to camera</Text>
                        </View>
                      </Callout>
                    </Marker>
                  )
                })
              }
            </MapView>
            :
            <View style={{ flex: 3 }}>
              <View style={styles.loading} >
                <Image style={styles.loadingGif} source={require('../../assets/loading.gif')} />
              </View>
            </View>
        }
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 3,
    width: '100%',
  },
  map: {
    width: '100%',
    flex: 3,
  },
  imageBackground: {
    width: '100%',
    display: 'flex',
    flexDirection: 'column',
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1.5,
    position: 'relative',
  },
  landmarkInfo: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    backgroundColor: 'white',
    borderRadius: 15,
    padding: 10,
    fontSize: 14
  },
  overlay: {
    position: 'absolute',
    backgroundColor: 'black',
    opacity: 0.5,
    height: '100%',
    width: '100%'
  },
  bannerText: {
    color: 'white',
    fontSize: 17,
    fontWeight: '600',
    padding: 10
  },
  headerText: {
    color: 'white',
    fontSize: 22,
    fontWeight: '600',
    padding: 10,
  },
  pointsText: {
    color: 'white',
    fontSize: 16,
    fontWeight: '600',
    paddingLeft: 10,
  },
  flexRow: {
    display: 'flex',
    flexDirection: 'row'
  },
  loading: {
    flex: 3,
    position: 'absolute',
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    zIndex: 1,
    top: 0,
    left: 0,
    height: '100%',
    width: '100%',
    backgroundColor: 'black'
  },
  loadingGif: {
    position: 'relative',
    zIndex: 3,
  }
});

export default Home;