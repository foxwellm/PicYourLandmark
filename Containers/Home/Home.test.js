import React from 'react';
import { Home } from './Home';
import { shallow } from 'enzyme';
import { TouchableOpacity } from 'react-native';
import { Marker } from 'react-native-maps';

jest.mock('react-native-maps', () => {
  const React = require.requireActual('react');
  const MapView = require.requireActual('react-native-maps');

  class MockCallout extends React.Component {
    render() {
      return React.createElement('Callout', this.props, this.props.children);
    }
  }

  class MockMarker extends React.Component {
    render() {
      return React.createElement('Marker', this.props, this.props.children);
    }
  }

  class MockMapView extends React.Component {
    render() {
      return React.createElement('MapView', this.props, this.props.children);
    }
  }

  MockCallout.propTypes = MapView.Callout.propTypes;
  MockMarker.propTypes = MapView.Marker.propTypes;
  MockMapView.propTypes = MapView.propTypes;
  MockMapView.Marker = MockMarker;
  MockMapView.Callout = MockCallout;
  return MockMapView;
});

describe.skip('Home', () => {

  let wrapper;
  let mockLatitude;
  let mockLongitude;
  let mockChangeCurrentPage;

  beforeEach(() => {
    mockLatitude = 23.4;
    mockLongitude = 104.8;
    mockChangeCurrentPage = jest.fn();
    mockTakeLocationPhoto = jest.fn();
    wrapper = shallow(<Home currentLatitude={mockLatitude} currentLongitude={mockLongitude} changeCurrentPage={mockChangeCurrentPage} takeLocationPhoto={mockTakeLocationPhoto} />);
  });

  it('should match the snapshot when a user has not clicked on a location', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('when the user clicks a location that they have already visited and are close enough to retake a photo', () => {
    beforeEach(() => {
      wrapper.setState({ clickedLocation: true, selectedVisited: true, calculatedDistance: 0 });
    });

    it('should match the snapshot when the user clicks a location', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call takeProfilePic when user does not have a profile picture', () => {
      wrapper.find(TouchableOpacity).first().simulate('press');
      expect(mockChangeCurrentPage).toHaveBeenCalledWith('Camera');
    });
  });

  it('when the user clicks a location that they have already visited but they are not close enough to retake a photo', () => {
    wrapper.setState({ clickedLocation: true, selectedVisited: true, calculatedDistance: 2 });
    expect(wrapper).toMatchSnapshot();
  });

  describe('when a user clicks on a marker location but has not visited the location', () => {
    it('should match the snapshot if they are close enough to take a photo', () => {
      wrapper.setState({ clickedLocation: true, selectedVisited: false, calculatedDistance: 0 });
      wrapper.find(TouchableOpacity).first().simulate('press');
      expect(mockChangeCurrentPage).toHaveBeenCalledWith('Camera');
    });

    it('should match the snapshot if the user is too far away to take a photo', () => {
      wrapper.setState({ clickedLocation: true, selectedVisited: false, calculatedDistance: 2 });
      expect(wrapper).toMatchSnapshot();
    });
  })

  it('should call calculateDistance when a marker is clicked', () => {
    wrapper.instance().calculateDistance = jest.fn();
    wrapper.find(Marker).first().simulate('press');
    expect(wrapper.instance().calculateDistance).toHaveBeenCalled();
  });

  describe('calculateDistance method', () => {
    const mockLat = 39.77;
    const mockLon = -104.99;
    const mockName = 'The Delectable Egg';
    const mockVisited = true;
    const mockDescription = 'Example description';
    const mockLocation = {
      lat: mockLat,
      lon: mockLon,
      name: mockName,
      visited: mockVisited,
      description: mockDescription
    }

    it('should take in a location and then setState with distance from current user and relevant location information', () => {
      wrapper.instance().calculateDistance(mockLocation)
      expect(wrapper.state('calculatedDistance')).toEqual('7668.42');
      expect(wrapper.state('selectedName')).toEqual(mockName);
      expect(wrapper.state('selectedVisited')).toEqual(mockVisited);
      expect(wrapper.state('selectedDescription')).toEqual(mockDescription);
      expect(wrapper.state('clickedLocation')).toEqual(true);
    })
  })
});
