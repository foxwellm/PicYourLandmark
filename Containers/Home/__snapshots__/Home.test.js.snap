// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should match the snapshot when a user has not clicked on a location 1`] = `
<Component
  style={
    Object {
      "flex": 3,
      "width": "100%",
    }
  }
>
  <ImageBackground
    source={1}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1.5,
        "flexDirection": "column",
        "justifyContent": "center",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "black",
          "height": "100%",
          "opacity": 0.5,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
    <Component
      style={
        Object {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
          "padding": 10,
        }
      }
    >
      Earn gems by taking and uploading pics of you at various locations
    </Component>
    <Themed.Icon
      color="white"
      name="camera-retro"
      size={40}
      type="font-awesome"
    />
  </ImageBackground>
  <MockMapView
    initialRegion={
      Object {
        "latitude": 23.4,
        "latitudeDelta": 0.015,
        "longitude": 104.8,
        "longitudeDelta": 0.0121,
      }
    }
    provider="google"
    showsUserLocation={true}
    style={
      Object {
        "flex": 3,
        "width": "100%",
      }
    }
  >
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.75302,
          "longitude": -104.9965,
        }
      }
      key="1"
      onPress={[Function]}
      pinColor="#000FFF"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Summit
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            example 1 description
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.75023,
          "longitude": -104.9965,
        }
      }
      key="2"
      onPress={[Function]}
      pinColor="#FFF000"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            The Delectable Egg
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            example 2 description
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.77023,
          "longitude": -104.9965,
        }
      }
      key="3"
      onPress={[Function]}
      pinColor="#000FFF"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Far away example
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            over a mile away
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
  </MockMapView>
</Component>
`;

exports[`Home when a user clicks on a marker location but has not visited the location should match the snapshot if the user is too far away to take a photo 1`] = `
<Component
  style={
    Object {
      "flex": 3,
      "width": "100%",
    }
  }
>
  <ImageBackground
    source={1}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1.5,
        "flexDirection": "column",
        "justifyContent": "center",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "black",
          "height": "100%",
          "opacity": 0.5,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Component
        style={
          Object {
            "color": "white",
            "fontSize": 22,
            "fontWeight": "600",
            "padding": 10,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Themed.Icon
        color="#f44336"
        name="diamond"
        size={15}
        type="font-awesome"
      />
      <Component
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingLeft": 10,
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        10
         gems
      </Component>
    </Component>
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Themed.Icon
        color="#00bcd4"
        name="car"
        size={15}
        type="font-awesome"
      />
      <Component
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingLeft": 10,
          }
        }
      >
        2
         mi. away
      </Component>
    </Component>
    <Component>
      <Component
        style={
          Object {
            "color": "white",
            "fontSize": 16,
            "fontWeight": "600",
            "paddingLeft": 10,
          }
        }
      >
        Get within 1 mile of the landmark to take a photo and add it to your collection
      </Component>
    </Component>
  </ImageBackground>
  <MockMapView
    initialRegion={
      Object {
        "latitude": 23.4,
        "latitudeDelta": 0.015,
        "longitude": 104.8,
        "longitudeDelta": 0.0121,
      }
    }
    provider="google"
    showsUserLocation={true}
    style={
      Object {
        "flex": 3,
        "width": "100%",
      }
    }
  >
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.75302,
          "longitude": -104.9965,
        }
      }
      key="1"
      onPress={[Function]}
      pinColor="#000FFF"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Summit
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            example 1 description
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.75023,
          "longitude": -104.9965,
        }
      }
      key="2"
      onPress={[Function]}
      pinColor="#FFF000"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            The Delectable Egg
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            example 2 description
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.77023,
          "longitude": -104.9965,
        }
      }
      key="3"
      onPress={[Function]}
      pinColor="#000FFF"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Far away example
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            over a mile away
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
  </MockMapView>
</Component>
`;

exports[`Home when the user clicks a location that they have already visited and are close enough to retake a photo should match the snapshot when the user clicks a location 1`] = `
<Component
  style={
    Object {
      "flex": 3,
      "width": "100%",
    }
  }
>
  <ImageBackground
    source={1}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1.5,
        "flexDirection": "column",
        "justifyContent": "center",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "black",
          "height": "100%",
          "opacity": 0.5,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Component
        style={
          Object {
            "color": "white",
            "fontSize": 22,
            "fontWeight": "600",
            "padding": 10,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Themed.Icon
        color="#f44336"
        name="diamond"
        size={15}
        type="font-awesome"
      />
      <Component
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingLeft": 10,
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        10
         gems
      </Component>
    </Component>
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Themed.Icon
        color="#00bcd4"
        name="car"
        size={15}
        type="font-awesome"
      />
      <Component
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingLeft": 10,
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        0
         mi. away 
      </Component>
    </Component>
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <TouchableOpacity
        activeOpacity={0.2}
        onPress={[Function]}
      >
        <Themed.Icon
          color="#4caf50"
          name="camera"
          size={15}
          type="font-awesome"
        />
        <Component
          style={
            Array [
              Object {
                "color": "white",
                "fontSize": 16,
                "fontWeight": "600",
                "paddingLeft": 10,
              },
            ]
          }
        >
          Retake Photo
        </Component>
      </TouchableOpacity>
    </Component>
  </ImageBackground>
  <MockMapView
    initialRegion={
      Object {
        "latitude": 23.4,
        "latitudeDelta": 0.015,
        "longitude": 104.8,
        "longitudeDelta": 0.0121,
      }
    }
    provider="google"
    showsUserLocation={true}
    style={
      Object {
        "flex": 3,
        "width": "100%",
      }
    }
  >
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.75302,
          "longitude": -104.9965,
        }
      }
      key="1"
      onPress={[Function]}
      pinColor="#000FFF"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Summit
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            example 1 description
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.75023,
          "longitude": -104.9965,
        }
      }
      key="2"
      onPress={[Function]}
      pinColor="#FFF000"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            The Delectable Egg
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            example 2 description
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.77023,
          "longitude": -104.9965,
        }
      }
      key="3"
      onPress={[Function]}
      pinColor="#000FFF"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Far away example
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            over a mile away
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
  </MockMapView>
</Component>
`;

exports[`Home when the user clicks a location that they have already visited but they are not close enough to retake a photo 1`] = `
<Component
  style={
    Object {
      "flex": 3,
      "width": "100%",
    }
  }
>
  <ImageBackground
    source={1}
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flex": 1.5,
        "flexDirection": "column",
        "justifyContent": "center",
        "position": "relative",
        "width": "100%",
      }
    }
  >
    <Component
      style={
        Object {
          "backgroundColor": "black",
          "height": "100%",
          "opacity": 0.5,
          "position": "absolute",
          "width": "100%",
        }
      }
    />
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Component
        style={
          Object {
            "color": "white",
            "fontSize": 22,
            "fontWeight": "600",
            "padding": 10,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Themed.Icon
        color="#f44336"
        name="diamond"
        size={15}
        type="font-awesome"
      />
      <Component
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingLeft": 10,
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        10
         gems
      </Component>
    </Component>
    <Component
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
        }
      }
    >
      <Themed.Icon
        color="#00bcd4"
        name="car"
        size={15}
        type="font-awesome"
      />
      <Component
        style={
          Array [
            Object {
              "color": "white",
              "fontSize": 16,
              "fontWeight": "600",
              "paddingLeft": 10,
            },
            Object {
              "color": "white",
            },
          ]
        }
      >
        2
         mi. away 
      </Component>
    </Component>
  </ImageBackground>
  <MockMapView
    initialRegion={
      Object {
        "latitude": 23.4,
        "latitudeDelta": 0.015,
        "longitude": 104.8,
        "longitudeDelta": 0.0121,
      }
    }
    provider="google"
    showsUserLocation={true}
    style={
      Object {
        "flex": 3,
        "width": "100%",
      }
    }
  >
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.75302,
          "longitude": -104.9965,
        }
      }
      key="1"
      onPress={[Function]}
      pinColor="#000FFF"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Summit
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            example 1 description
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.75023,
          "longitude": -104.9965,
        }
      }
      key="2"
      onPress={[Function]}
      pinColor="#FFF000"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            The Delectable Egg
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            example 2 description
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
    <MockMarker
      coordinate={
        Object {
          "latitude": 39.77023,
          "longitude": -104.9965,
        }
      }
      key="3"
      onPress={[Function]}
      pinColor="#000FFF"
    >
      <MockCallout
        tooltip={true}
      >
        <Component
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "borderRadius": 15,
              "display": "flex",
              "flexDirection": "column",
              "fontSize": 14,
              "padding": 10,
            }
          }
        >
          <Component
            style={
              Object {
                "fontWeight": "bold",
              }
            }
          >
            Far away example
          </Component>
          <Component
            style={
              Object {
                "fontStyle": "italic",
              }
            }
          >
            over a mile away
          </Component>
          <Component>
            Distance away: mi
          </Component>
          <Component>
            Visited: no
          </Component>
          <Component>
            Link - takes to camera
          </Component>
        </Component>
      </MockCallout>
    </MockMarker>
  </MockMapView>
</Component>
`;
